trl/trainer/kto_trainer.py

```diff
@@ -360,15 +360,15 @@ def make_inputs_require_grad(module, input, output):
             # merge the datasets
             eval_dataset = concatenate_datasets([eval_dataset, eval_KL_dataset], axis=1)
 
-        desirable = train_dataset.filter(lambda x: x["label"])
-        undesirable = train_dataset.filter(lambda x: not x["label"])
+        num_desirable = sum(train_dataset["label"])
+        num_undesirable = len(train_dataset) - num_desirable
 
-        if len(desirable) != len(undesirable):
+        if num_desirable != num_undesirable:
             # The lower and upper bounds come from Eq. (8) of https://arxiv.org/abs/2402.01306
-            des_weight_lower_bound = round((len(undesirable) * self.undesirable_weight / len(desirable)) * 1, 2)
-            des_weight_upper_bound = round((len(undesirable) * self.undesirable_weight / len(desirable)) * 1.33, 2)
-            und_weight_lower_bound = round((len(desirable) * self.desirable_weight / len(undesirable)) / 1.33, 2)
-            und_weight_upper_bound = round((len(desirable) * self.desirable_weight / len(undesirable)) / 1, 2)
+            des_weight_lower_bound = round((len(num_undesirable) * self.undesirable_weight / len(num_desirable)) * 1, 2)
```

Comment:
isn´t `num_undesirable ` the int length already? For me `len(num_undesirable)` throws an error